src/redturtle/prenotazioni/upgrades.py

```diff
@@ -211,3 +214,106 @@ def to_1500(context):
+        "notify_on_confirm_message": (
+            "La prenotazione ${booking_type} per ${title} è stata confermata!"
+            "Se non hai salvato o stampato il promemoria, puoi visualizzarlo su <a href=${booking_print_url}>questo link</a>"
+            "Se desideri cancellare la prenotazione, accedi a <a href=${booking_print_url}>questo link</a>"
+        ),
+        "notify_on_move_subject": ("Modifica data di prenotazione per ${title}"),
+        "notify_on_move_message": (
+            "L'orario della sua prenotazione ${booking_type} è stata modificato."
+            "La nuova data è ${booking_date} alle ore ${booking_time}."
+            "Controlla o stampa il nuovo promemoria su  <a href=${booking_print_url}>questo link</a>."
+        ),
+        "notify_on_refuse_subject": "Prenotazione rifiutata per ${title}",
+        "notify_on_refuse_message": "La prenotazione ${booking_type} del ${booking_date} delle ore ${booking_time} è stata rifiutata.",
+    }
+
+    for brain in api.portal.get_tool("portal_catalog")(
+        portal_type="PrenotazioniFolder"
+    ):
+        object = brain.getObject()
+
+        logger.info(f"[1501-1502] Updating fields on {brain.getPath()}")
+
+        object.templates_usage = RichTextValue(
+            instructions_default, "text/html", "text/html"
+        )
+
+        for name, value in new_fields.items():
+            setattr(object, name, value)
+
+
+def to_1502_upgrade_contentrules(context):
+    from plone.contentrules.engine.interfaces import IRuleStorage
+    from plone.contentrules.engine.interfaces import IRuleAssignmentManager
+
+    rules_to_delete = [
+        "booking-accepted",
+        "booking-moved",
+        "booking-created-user",
+        "booking-refuse",
+        "booking-confirm",
+    ]
+
+    rule_storage = getUtility(IRuleStorage)
+
+    for brain in api.portal.get_tool("portal_catalog")(
+        portal_type="PrenotazioniFolder"
+    ):
+        obj = brain.getObject()
+        assignable = IRuleAssignmentManager(obj, None)
+        contentrules_mapping = {
+            "booking-accepted": "notify_on_submit",
+            "booking-moved": "notify_on_move",
+            "booking-refuse": "notify_on_refuse",
+        }
+
+        for old, new in contentrules_mapping.items():
+            if assignable.get("booking-confirm", None):
+                setattr(obj, "notify_on_confirm", True)
+                setattr(obj, "auto_confirm", True)
+                continue
+            if assignable.get(old, None):
+                setattr(obj, new, True)
+
+    for rule in rules_to_delete:
+        if rule_storage.get(rule):
+            # It is supposed that all the rule assignments will be deleted by plone.app.contentrules
+            # event handlers which are supposed to do that
+            logger.info(f"[1501-1502] Deleting contentrule `{rule}`")
+            del rule_storage[rule]
```

Comment:
probably you should do it also on root and on folders

Reply:
solved